utils/index.d.ts

```diff
@@ -176,12 +177,157 @@ export interface ISubscriptionContext {
+     */
+    commandArgs?: unknown[];
+    contextValue: string;
+
+    /**
+      * Implement this to display child resources. Should not be called directly
+      * @param clearCache If true, you should start the "Load more..." process over
+      * @param context The action context
+      */
+    loadMoreChildrenImpl?(clearCache: boolean, context: IActionContext): Promise<AzExtTreeItem[]>;
+
+    /**
+    * Implement this as a part of the "Load more..." action. Should not be called directly
+    * @returns 'true' if there are more children and a "Load more..." node should be displayed
+    */
+    hasMoreChildrenImpl?(): boolean;
+
+    /**
+     * Implement this if you want the 'create' option to show up in the tree picker. Should not be called directly
+     * @param context The action context and any additional user-defined options that are passed to the `AzExtParentTreeItem.createChild` or `AzExtTreeDataProvider.showTreeItemPicker`
+     */
+    createChildImpl?(context: ICreateChildImplContext): Promise<AzExtTreeItem>;
+
+    /**
+     * Override this if you want non-default (i.e. non-alphabetical) sorting of children. Should not be called directly
+     * @param item1 The first item to compare
+     * @param item2 The second item to compare
+     * @returns A negative number if the item1 occurs before item2; positive if item1 occurs after item2; 0 if they are equivalent
+     */
+    compareChildrenImpl?(item1: AzExtTreeItem, item2: AzExtTreeItem): number;
+
+    /**
+    * If this treeItem should not show up in the tree picker or you want custom logic to show quick picks, implement this to provide a child that corresponds to the expectedContextValue. Should not be called directly
+    * Otherwise, all children will be shown in the tree picker
+    */
+    pickTreeItemImpl?(expectedContextValues: (string | RegExp)[], context: IActionContext): AzExtTreeItem | undefined | Promise<AzExtTreeItem | undefined>;
+
+    /**
+     * Implement this to support the 'delete' action in the tree. Should not be called directly
+     */
+    deleteTreeItemImpl?(context: IActionContext): Promise<void>;
+
+    /**
+     * Implement this to execute any async code when this node is refreshed. Should not be called directly
+     */
+    refreshImpl?(context: IActionContext): Promise<void>;
+
+    /**
+     * Optional function to filter items displayed in the tree picker. Should not be called directly
+     * If not implemented, it's assumed that 'isAncestorOf' evaluates to true
+     */
+    isAncestorOfImpl?(contextValue: string | RegExp): boolean;
+
+    /**
+     * If implemented, resolves the tooltip at the time of hovering, and the value of the `tooltip` property is ignored. Otherwise, the `tooltip` property is used.
+     */
+    resolveTooltip?(): Promise<string | MarkdownString>;
+}
+
+export type IAzExtTreeItem = AbstractAzExtTreeItem & SealedAzExtTreeItem;
```

Comment:
I know we generally avoid the `IFoo` names but `AzExtTreeItem` is taken

Reply:
ISoundsGood